framework/include/reporters/ReporterContext.h

```diff
@@ -630,6 +748,36 @@ class ReporterVectorContext : public ReporterContext<std::vector<T>>
    */
   virtual void resize(dof_id_type local_size) override { this->_state.value().resize(local_size); }
 
+  /**
+   * Since we know that the _state value is a vector type, we can clear it.
+   */
+  virtual void clear() override { this->_state.value().clear(); }
+
+  virtual void vectorSum() override
+  {
+    // Case 1: T is type that we can sum
+    if constexpr (std::is_arithmetic<T>::value &&
+                  !std::is_same<T, bool>::value) // We can't sum bools.
+    {
+      this->comm().sum(this->_state.value());
+      return;
+    }
+    // Case 2: T is a vector
+    else if constexpr (is_std_vector<T>::value)
+    {
+      using ValueType = typename T::value_type;
+      // Check if the ValueType is a vector
+      if constexpr (std::is_arithmetic<ValueType>::value && !std::is_same<ValueType, bool>::value)
+        for (auto & val_vec : this->_state.value())
```

Comment:
I think we need to make sure the vecotr is of the same size everywhere otherwise this will hang